RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -223,11 +222,17 @@ void Event::addPixelSegmentToEvent(std::vector<unsigned int> const& hitIndices0,
 
     nTotalSegments_ += n_max_pixel_segments_per_module;
 
-    segmentsInGPU_.emplace();
-    segmentsBuffers_.emplace(nTotalSegments_, nLowerModules_, n_max_pixel_segments_per_module, devAcc_, queue_);
-    segmentsInGPU_->setData(*segmentsBuffers_);
-
-    alpaka::memcpy(queue_, segmentsBuffers_->nMemoryLocations_buf, nTotalSegments_view);
+    std::array<int, 3> const segments_sizes{{static_cast<int>(nTotalSegments_),
+                                             static_cast<int>(nLowerModules_ + 1),
+                                             static_cast<int>(n_max_pixel_segments_per_module)}};
+    segmentsDC_.emplace(segments_sizes, queue_);
+
+    auto nSegments_view =
+        alpaka::createView(devAcc_, segmentsDC_->view<SegmentsOccupancySoA>().nSegments(), nLowerModules_ + 1);
```

Comment:
```suggestion
        alpaka::createView(devAcc_, segmentsDC_->view<SegmentsOccupancySoA>().nSegments(), segments_sizes[1]);
```

same below

Reply:
Sorry I missed these. I'll fix them and finish the SoA migration next week.